index.js

```diff
@@ -174,6 +189,10 @@ Filter.prototype.build = function() {
 
       return result;
     }));
+  }).then(function(completedResult) {
+    return allSettled(pendingWork).then(function() {
+      return Promise.all(completedResult.concat(pendingWork));
```

Comment:
I've tested this I don't see that it's hiding or suppressing any errors. I have the `'completes all pending work before returning'` test which checks that the build fails and there is an error:

```javascript
try {
  yield output.build();
} catch(error) {
  didFail = true;
  expect(error.message).to.contain('file failed for some reason', 'error message text should match');
}
expect(didFail).to.eql(true, 'build should fail');
```

Since `completedResult` and `pendingWork` are arrays of Promises, this retains any errors/rejections from those:

```javascript
return Promise.all(completedResult.concat(pendingWork));
```
When testing I see that the error is being thrown from `processAndCacheFile()`

If I change this to
```javascipt
return completedResult.concat(pendingWork);
```

That hides the errors and shows the build as successful when it's not.